lib/knetwork/nopclose.go

```diff
@@ -0,0 +1,35 @@
+package knetwork
+
+import (
+	"io"
+)
+
+type nopReadCloser struct {
+	io.Reader
+}
+
+func (n *nopReadCloser) Close() error {
+	return nil
+}
+
+// NopReadCloser turns any io.Reader into a io.ReadCloser with a nop closer.
+//
+// This can also be used to ignore Close() events on another ReadCloser.
+func NopReadCloser(reader io.Reader) io.ReadCloser {
+	return &nopReadCloser{reader}
+}
+
+type nopWriteCloser struct {
```

Comment:
So I don't like this. It scares me and NoOp writers scare me. It has it's valid uses, but I would rather see the workarounds in the proposal thread that wanted this feature.

https://github.com/golang/go/issues/22823

Reply:
Otherwise, yolo.